cdp/smart_contract.py

```diff
@@ -0,0 +1,324 @@
+import json
+import time
+from decimal import Decimal
+from typing import Any
+
+from eth_account.signers.local import LocalAccount
+
+from cdp.cdp import Cdp
+from cdp.client.models.create_smart_contract_request import CreateSmartContractRequest
+from cdp.client.models.deploy_smart_contract_request import DeploySmartContractRequest
+from cdp.client.models.smart_contract import SmartContract as SmartContractModel
+from cdp.client.models.smart_contract_type import SmartContractType as SmartContractTypeModel
+from cdp.transaction import Transaction
+
+
+class SmartContract:
+    """A representation of a SmartContract on the blockchain."""
+
+    class Type:
+        """Enumeration of SmartContract types."""
+
+        ERC20 = "ERC20"
+        ERC721 = "ERC721"
+        ERC1155 = "ERC1155"
+
+        def __str__(self) -> str:
+            """Return a string representation of the Type."""
+            return self.value
+
+        def __repr__(self) -> str:
+            """Return a string representation of the Type."""
+            return str(self)
+
+    class TokenOptions:
+        """Options for ERC20 token contracts."""
+
+        def __init__(self, name: str, symbol: str, total_supply: int | Decimal | str):
+            self.name = name
+            self.symbol = symbol
+            self.total_supply = total_supply
+
+    class NFTOptions:
+        """Options for ERC721 NFT contracts."""
+
+        def __init__(self, name: str, symbol: str, base_uri: str):
+            self.name = name
+            self.symbol = symbol
+            self.base_uri = base_uri
+
+    class MultiTokenOptions:
+        """Options for ERC1155 multi-token contracts."""
+
+        def __init__(self, uri: str):
+            self.uri = uri
```

Comment:
Seems like these should just be typed dictionaries since they are only used for configuration?